build-tests/heft-example-plugin-01/src/index.ts

```diff
@@ -0,0 +1,36 @@
+import { SyncHook } from 'tapable';
+import {
+  IHeftPlugin,
+  HeftSession,
+  HeftConfiguration,
+  IBuildStageContext,
+  IPreCompileSubstage
+} from '@rushstack/heft';
+
+const PLUGIN_NAME: string = 'example-plugin-01';
+
+export interface IExamplePlugin01Hooks {
+  exampleHook: SyncHook;
+}
+
+export class ExamplePlugin01 implements IHeftPlugin {
+  public displayName: string = PLUGIN_NAME;
+
+  public hooks: IExamplePlugin01Hooks;
+
+  public apply(heftSession: HeftSession, heftConfiguration: HeftConfiguration): void {
+    this.hooks = {
+      exampleHook: new SyncHook()
+    };
+
+    heftSession.hooks.build.tap(PLUGIN_NAME, (build: IBuildStageContext) => {
+      build.hooks.preCompile.tap(PLUGIN_NAME, (preCompile: IPreCompileSubstage) => {
+        preCompile.hooks.run.tap(PLUGIN_NAME, () => {
+          this.hooks.exampleHook.call();
+        });
+      });
+    });
+  }
+}
+
+export default new ExamplePlugin01();
```

Comment:
`export default` is deprecated because it does not enforce the name of the exported thing.

Might be better for the contract to use a name like:
```ts
export const heftPlugin = new ExamplePlugin01();
```

For API Documenter plugins we export a more rich manifest contract [like this](https://github.com/microsoft/rushstack/blob/5627415032539c27e93fdddac4edf93d2c87a311/repo-scripts/doc-plugin-rush-stack/src/index.ts#L7):
```ts
export const apiDocumenterPluginManifest: IApiDocumenterPluginManifest = {
  manifestVersion: 1000,
  features: [
    {
      featureName: 'rush-stack-markdown-documenter',
      kind: 'MarkdownDocumenterFeature',
      subclass: RushStackFeature
    }
  ]
};
```

Reply:
BTW I'm planning to enable a lint rule to ban `export default` because I keep finding code that is using it.